plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java

```diff
@@ -121,4 +182,198 @@ public void testChunkSize() {
+
+                    final StringBuilder list = new StringBuilder();
+                    list.append("{\"kind\":\"storage#objects\",\"items\":[");
+                    for (Iterator<Map.Entry<String, BytesReference>> it = listOfBlobs.iterator(); it.hasNext(); ) {
+                        Map.Entry<String, BytesReference> blob = it.next();
+                        list.append("{\"kind\":\"storage#object\",");
+                        list.append("\"bucket\":\"bucket\",");
+                        list.append("\"name\":\"").append(blob.getKey()).append("\",");
+                        list.append("\"id\":\"").append(blob.getKey()).append("\",");
+                        list.append("\"size\":\"").append(blob.getValue().length()).append("\"");
+                        list.append('}');
+
+                        if (it.hasNext()) {
+                            list.append(',');
+                        }
+                    }
+                    list.append("]}");
+
+                    byte[] response = list.toString().getBytes(UTF_8);
+                    exchange.getResponseHeaders().add("Content-Type", "application/json; charset=utf-8");
+                    exchange.sendResponseHeaders(RestStatus.OK.getStatus(), response.length);
+                    exchange.getResponseBody().write(response);
+
+                } else if (Regex.simpleMatch("GET /storage/v1/b/bucket*", request)) {
+                    byte[] response = ("{\"kind\":\"storage#bucket\",\"name\":\"bucket\",\"id\":\"0\"}").getBytes(UTF_8);
+                    exchange.getResponseHeaders().add("Content-Type", "application/json; charset=utf-8");
+                    exchange.sendResponseHeaders(HttpStatus.SC_OK, response.length);
+                    exchange.getResponseBody().write(response);
+
+                } else if (Regex.simpleMatch("GET /download/storage/v1/b/bucket/o/*", request)) {
+                    BytesReference blob = blobs.get(exchange.getRequestURI().getPath().replace("/download/storage/v1/b/bucket/o/", ""));
+                    if (blob != null) {
+                        exchange.getResponseHeaders().add("Content-Type", "application/octet-stream");
+                        exchange.sendResponseHeaders(RestStatus.OK.getStatus(), blob.length());
+                        exchange.getResponseBody().write(blob.toBytesRef().bytes);
+                    } else {
+                        exchange.sendResponseHeaders(RestStatus.NOT_FOUND.getStatus(), -1);
+                    }
+
+                } else if (Regex.simpleMatch("DELETE /storage/v1/b/bucket/o/*", request)) {
+                    int deletions = 0;
+                    for (Iterator<Map.Entry<String, BytesReference>> iterator = blobs.entrySet().iterator(); iterator.hasNext(); ) {
+                        Map.Entry<String, BytesReference> blob = iterator.next();
+                        if (blob.getKey().startsWith(exchange.getRequestURI().toString())) {
```

Comment:
Why `startsWith` and not `equals` here? (there's not glob delete in GCS + it seems these tests pass fine with equals?)

Reply:
I guess it's a leftover, I don't see any reasons to use `startsWith`. I pushed aef77214e456d0217a1ef7ff95ed11cefcb0b552